Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,208 @@
+  http://www.jstor.org/stable/27958492. Accessed 21 Feb. 2024.
+
+## Tags
+
+abundant, deficient, weird, pseudoperfect
+-/
+
+open BigOperators Finset
+
+namespace Nat
+
+variable {n m : ℕ}
+
+/-- `n : ℕ` is _abundant_ if and only if the sum of the proper divisors of `n` is greater than `n`
+  -/
+def Abundant (n : ℕ) : Prop := ∑ i in properDivisors n, i > n
+
+/-- `n : ℕ` is _deficient_ if and only if the sum of the proper divisors of `n` is less than `n` -/
+def Deficient (n : ℕ) : Prop := ∑ i in properDivisors n, i < n
+
+/-- `n : ℕ` is _pseudoperfect_ if and only if there exists a subset of the proper divisors of n
+  such that the sum of that subset is equal to `n` and `n` is positive -/
+def Pseudoperfect (n : ℕ) : Prop :=
+  0 < n ∧ ∃ s : Finset ℕ, s ∈ powerset (properDivisors n) ∧ ∑ i in s, i = n
+
+/-- `n : ℕ` is a _weird_ number if and only if it is both abundant and pseudoperfect -/
+def WeirdNumber (n : ℕ) : Prop := Abundant n ∧ ¬ Pseudoperfect n
+
+theorem not_pseudoperfect_iff_forall :
+    ¬ Pseudoperfect n ↔ n = 0 ∨ ∀ (s : Finset ℕ), s ∈ powerset (properDivisors n)
+      → ∑ i in s, i ≠ n := by
+  rw [Pseudoperfect, not_and_or]
+  simp only [not_lt, nonpos_iff_eq_zero, mem_powerset, not_exists, not_and, ne_eq]
+
+theorem one_deficient : Deficient 1 := le.refl
+
+theorem two_deficient : Deficient 2 := le.refl
+
+theorem three_deficient : Deficient 3 := by
+  dsimp only [Deficient]
+  norm_num
+
+theorem six_perfect : Perfect 6 := by
+  dsimp only [Perfect]
+  rw [show properDivisors 6 = {1, 2, 3} by rfl]
+  norm_num
+
+theorem twelve_abundant : Abundant 12 := by
+  dsimp only [Abundant]
+  rw [show properDivisors 12 = {1,2,3,4,6} by rfl]
+  norm_num
+
+set_option maxRecDepth 2000 in
+theorem seventy_weird : WeirdNumber 70 := by
+  dsimp only [WeirdNumber, Abundant]
+  rw [not_pseudoperfect_iff_forall]
+  have div70 : properDivisors 70 = {1, 2, 5, 7, 10, 14, 35} := by rfl
+  constructor
+  · rw [div70]
+    repeat norm_num
+  · rw [div70]
+    right
+    intro s hs
+    fin_cases hs <;> decide
+
+lemma deficient_not_abundant_or_perfect (hn : 0 < n) :
+    Deficient n ↔ ¬ Abundant n ∧ ¬ Perfect n := by
+    dsimp only [Perfect, Abundant, Deficient]
+    omega
+
+lemma perfect_not_abundant_or_deficient (hn : 0 < n) :
+    Perfect n ↔ ¬ Abundant n ∧ ¬ Deficient n := by
+    dsimp only [Perfect, Abundant, Deficient]
+    omega
+
+lemma abundant_not_perfect_or_deficient (hn : 0 < n) :
+    Abundant n ↔ ¬ Perfect n ∧ ¬ Deficient n := by
+    dsimp only [Perfect, Abundant, Deficient]
+    omega
+
+/-- A positive natural number is either deficient, perfect, or abundant -/
+theorem deficient_or_perfect_or_abundant (hn : 0 < n) :
+    Deficient n ∨ Abundant n ∨ Perfect n := by
+    dsimp only [Perfect, Abundant, Deficient]
+    omega
+
+theorem perfect_pseudoperfect (h : Perfect n) : Pseudoperfect n :=
+  ⟨h.2, _, mem_powerset_self _, h.1⟩
+
+theorem prime_not_abundant (h : Prime n) : ¬ Abundant n :=
```

Comment:
For readability I'd be tempted to use `p` for `n` when `p` is prime.

Reply:
Okay I can implement that

Reply:
Looks like you didn't implement that?